### DISCUSSION TITLE
ESP32-C3 fatal error on installing micropython

### DISCUSSION THREAD

**Discussion by ctvarlan** [micropython/micropython#9881] [Hardware & Peripherals]
Hello everyone,
The first try to put micropython on an older ESP32-C3. 
The command is:

_>esptool.py --chip esp32c3 --port COM4 --baud 460800 write_flash -z 0x0 esp32c3-usb-20220117-v1.18.bin_

and the messages are:

_esptool.py v4.3
Serial port COM4
Connecting....
Chip is ESP32-C3 (revision v0.2)
Features: WiFi, BLE
Crystal is 40MHz
MAC: 7c:df:a1:43:10:58
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 460800
Changed.
Configuring flash size..._

_A fatal error occurred: esp32c3-usb-20220117-v1.18.bin requires chip revision 3 or higher (this chip is revision 2). Use --force to flash anyway._

I tried --force and the result is the same.

What should I do?

Thank you for your attention,

**Comment by puppet13th** [comment]
Hi there. What board are you using now? Have you tried using older esp tool? Try using version 3.3

**Comment by ctvarlan** [comment]
Hi,
Thank you for your response.

My board is ESP32-C3-DevKitM-1.

I just installed esptool 3.3, and I'll give it a try. I let you know how it
goes.

Thank you,

**Comment by ctvarlan** [comment]
I gave it a try with esptool 3.3. The flashing was smooth, but the surprise
came after.
At boot (I'm using Thonny), there is an infinite loop with this content:

Build:Sep 18 2020
rst:0x3 (RTC_SW_SYS_RST),boot:0xc (SPI_FAST_FLASH_BOOT)
Saved PC:0x403d1232
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fcd6100,len:0xe3c
load:0x403ce000,len:0x6dc
load:0x403d0000,len:0x28ac
entry 0x403ce000
E (34)  ESP-ROM:esp32c3-20200918
Build:Sep 18 2020
rst:0x3 (RTC_SW_SYS_RST),boot:0xc (SPI_FAST_FLASH_BOOT)
Saved PC:0x403d1232
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fcd6100,len:0xe3c
load:0x403ce000,len:0x6dc
load:0x403d0000,len:0x28ac
entry 0x403ce000
E (34)  ESP-ROM:esp32c3-20200918

Any clue?
Thanks a lot,

**Comment by yn386** [comment]
Hi, there.
Can you build micropython from source code on Ubuntu?
If yes, try following steps may help you:

1. Checkout micropython v1.18.
```
git clone https://github.com/micropython/micropython.git -b v1.18
```

2. Setup esp-idf with ci.sh.
```
cd micropython/tools
. ./ci.sh
ci_esp32_idf44_setup
. ./esp-idf/export.sh
```

3. Apply attached patch [v1.18-for-esp32c3-rev-2.zip](https://github.com/micropython/micropython/files/10012589/v1.18-for-esp32c3-rev-2.zip). This patch changes required minimum revision 3 to 2.
```
cd ..
unzip v1.18-for-esp32c3-rev-2.zip
patch -p1 < v1.18-for-esp32c3-rev-2.patch
```

4. Build micropython.
```
make -C mpy-cross
cd ports/esp32
make submodules
make BOARD=GENERIC_C3_USB
```

5. Deploy micropython to your esp32c3.
```
make BOARD=GENERIC_C3_USB deploy PORT=/dev/ttyACM0
```

I'm sorry that I couldn't test this because I don't have ESP32C3 revision 2, but these steps worked with ESP32C3 revision 3.

**Reply by yn386** [reply]
If you want to use latest version of micropython, please replace step 1 and 2 with the following:

1. Checkout micropython.
```
git clone https://github.com/micropython/micropython.git
```
2. Setup esp-idf with ci.sh.
```
cd micropython/tools
. ./ci.sh
ci_esp32_setup_helper 142bb32c
. ./esp-idf/export.sh
```
Step 3 to 5 are the same as above.

**Reply by ctvarlan** [reply]
Hi yn386,
I'll give it a try, and it would be a completely new path for me, doing the build from scratch (source).
I hope it will work with Mint, I don't have Ubuntu.

Thank you and I will let you know,
